torch/csrc/inductor/aot_runtime/interface.h

```diff
@@ -64,58 +53,61 @@ AOTInductorError AOTInductorModelContainerDelete(
 // Runs the inference.
 AOTInductorError AOTInductorModelContainerRun(
     AOTInductorModelContainerHandle container_handle,
-    AOTInductorTensorHandle inputs_handle,
+    AOTInductorTensorHandle input_handles,
     size_t num_inputs,
-    AOTInductorTensorHandle outputs_handle,
+    AOTInductorTensorHandle output_handles,
     size_t num_outputs,
-    AOTInductorParamShape* output_shapes,
     AOTInductorStreamHandle stream_handle,
-    AOTInductorProxyExecutorHandle proxy_executor_handle);
+    AOTInductorProxyExecutorHandle proxy_executor_handle,
+    const int64_t** ret_output_sizes,
+    int64_t* ret_output_ndims);
 
 // Retrieves the number of inputs for the model.
 AOTInductorError AOTInductorModelContainerGetNumInputs(
     AOTInductorModelContainerHandle container_handle,
-    size_t* num_inputs_out);
+    size_t* ret_num_inputs);
 
 // Retrieves the input name at the given index.
 AOTInductorError AOTInductorModelContainerGetInputName(
     AOTInductorModelContainerHandle container_handle,
     size_t input_idx,
-    const char** input_name_out);
+    const char** ret_input_names);
 
 // Retrieves the input dtype at the given index.
 AOTInductorError AOTInductorModelContainerGetInputDtype(
     AOTInductorModelContainerHandle container_handle,
     size_t input_idx,
-    const char** input_name_out);
+    const char** ret_input_dtypes);
 
 // Retrieves the number of outputs for the model.
 AOTInductorError AOTInductorModelContainerGetNumOutputs(
     AOTInductorModelContainerHandle container_handle,
-    size_t* num_outputs_out);
+    size_t* ret_num_outputs);
 
 // Retrieves the output name at the given index.
 AOTInductorError AOTInductorModelContainerGetOutputName(
     AOTInductorModelContainerHandle container_handle,
     size_t output_idx,
-    const char** output_name_out);
+    const char** ret_output_names);
 
 // Retrieves the output dtype at the given index.
 AOTInductorError AOTInductorModelContainerGetOutputDtype(
     AOTInductorModelContainerHandle container_handle,
     size_t output_idx,
-    const char** output_name_out);
+    const char** ret_output_dtypes);
 
 // Retieves the input shape with the maximum dimension size for each dimension.
 AOTInductorError AOTInductorModelContainerGetMaxInputShape(
     AOTInductorModelContainerHandle container_handle,
     size_t input_idx,
-    AOTInductorParamShape* input_shape);
+    const int64_t** ret_input_sizes,
```

Comment:
This is a preexisting condition, but what's the ownership on the returned `int*`?

Reply:
It is borrow, so are other `ret_*`. I can add comments to state that.